identity_credential/src/revocation/status_list_2021/entry.rs

```diff
@@ -0,0 +1,136 @@
+    impl<'a> Visitor<'a> for ExactStrVisitor {
+      type Value = &'static str;
+      fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(formatter, "the exact string \"{}\"", self.0)
+      }
+      fn visit_str<E: Error>(self, str: &str) -> Result<Self::Value, E> {
+        if str == self.0 {
+          Ok(self.0)
+        } else {
+          Err(E::custom(format!("not \"{}\"", self.0)))
+        }
+      }
+    }
+
+    deserializer
+      .deserialize_str(ExactStrVisitor(CREDENTIAL_STATUS_TYPE))
+      .map(EntryType)
+  }
+}
+
+/// [StatusList2021Entry](https://www.w3.org/TR/2023/WD-vc-status-list-20230427/#statuslist2021entry) implementation.
+#[derive(Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct StatusList2021Entry {
+  id: Url,
+  #[serde(rename = "type")]
+  type_: EntryType,
+  status_purpose: StatusPurpose,
+  #[serde(deserialize_with = "serde_aux::prelude::deserialize_number_from_string")]
+  status_list_index: usize,
+  status_list_credential: Url,
+}
+
+impl TryFrom<&Status> for StatusList2021Entry {
+  type Error = serde_json::Error;
+  fn try_from(status: &Status) -> Result<Self, Self::Error> {
+    let json_status = serde_json::to_value(status).unwrap();
+    serde_json::from_value(json_status)
+  }
+}
+
+impl From<StatusList2021Entry> for Status {
+  fn from(entry: StatusList2021Entry) -> Self {
+    let json_status = serde_json::to_value(entry).unwrap(); // Safety: shouldn't go out of memory
+    serde_json::from_value(json_status).unwrap() // Safety: `StatusList2021Entry` is a credential status
+  }
+}
+
+impl StatusList2021Entry {
+  /// Creates a new [`StatusList2021Entry`].
+  pub fn new(status_list: Url, purpose: StatusPurpose, index: usize) -> Self {
+    let mut id = status_list.clone();
+    id.set_fragment(Some(format!("{index}").as_str()));
+
+    Self {
+      id,
+      type_: EntryType::default(),
+      status_purpose: purpose,
+      status_list_credential: status_list,
+      status_list_index: index,
+    }
+  }
+
+  /// Returns this `credentialStatus`'s `id`.
+  pub const fn id(&self) -> &Url {
+    &self.id
+  }
+
+  /// Returns the purpose of this entry.
+  pub const fn purpose(&self) -> StatusPurpose {
+    self.status_purpose
+  }
+
+  /// Returns the index of this entry.
+  pub const fn index(&self) -> usize {
+    self.status_list_index
+  }
+
+  /// Returns the referenced [`StatusList2021Credential`]'s [`Url`].
+  pub const fn credential(&self) -> &Url {
+    &self.status_list_credential
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  const STATUS_LIST_ENTRY_SAMPLE: &str = r#"
+{
+    "id": "https://example.com/credentials/status/3#94567",
+    "type": "StatusList2021Entry",
+    "statusPurpose": "revocation",
+    "statusListIndex": "94567",
+    "statusListCredential": "https://example.com/credentials/status/3"
+}"#;
+
+  #[test]
+  fn entry_deserialization_works() {
```

Comment:
It could be nice to check the rest of the values and add tests where the serialization fails, like when the `type` is wrong.